### PR TITLE
JUS-2178 add explicit commit mode, ability to subscribe from an offset

### PR DESCRIPTION
**Pull request by IAD**

### CODE REVIEW

File: .idea/copyright/AB.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<component name="CopyrightManager">
+  <copyright>
+    <option name="notice" value="Copyright (c) &amp;#36;{today.year} AccelByte Inc. All Rights Reserved.&#10;This is licensed software from AccelByte Inc, for limitations&#10;and restrictions contact your company contract manager." />
```
Comment (ajiyakin) [inline .idea/copyright/AB.xml:3]: Thanks for add this template :+1:

File: docker-compose-test.yml [hunk 2]
```diff
@@ -16,24 +16,30 @@
...
     ports:
       - "2181:2181"
     expose:
       - "2181"
+    networks:
+      - resource-network
 
   kafka:
-    image: wurstmeister/kafka
+    image: wurstmeister/kafka:2.12-2.5.0
```
Comment (ajiyakin) [inline docker-compose-test.yml:28]: This is some good practise, thank you for implementing this one :heart:

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Copyright 2019 AccelByte Inc
```
Comment (ajiyakin) [inline docker-compose.yml:1]: 2020?

File: eventstream.go [hunk 4]
```diff
@@ -260,6 +264,12 @@ func (s *SubscribeBuilder) Topic(topic string) *SubscribeBuilder {
 	return s
 }
 
+// Topic set topic that will be subscribe
```
Comment (ajiyakin) [inline eventstream.go:267]: Why the comment block says `Topic` while the func name is `Offset`?

File: kafka.go [hunk 5]
```diff
@@ -282,27 +286,69 @@ func (client *KafkaClient) Register(subscribeBuilder *SubscribeBuilder) error {
...
 	reader := kafka.NewReader(config)
 
 	go func() {
-		defer reader.Close() // nolint: errcheck
+		var eventProcessingFailed bool
+
+		defer func() {
+			if eventProcessingFailed {
+				if subscribeBuilder.ctx.Err() != nil {
+					// the subscription is shutting down. triggered by an external context cancellation
```
Comment (ajiyakin) [inline kafka.go:299]: Shouldn't we write to the log what happen here? Just for debugging. Thoughts?

Comment (IAD) [reply kafka.go:299]: @ajiyakin improved

**Pull request merged**